xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MUCPersistenceManager.java

```diff
@@ -147,6 +147,9 @@ public class MUCPersistenceManager {
         "INSERT INTO ofMucConversationLog (roomID,messageID,sender,nickname,logTime,subject,body,stanza) VALUES (?,?,?,?,?,?,?,?)";
     private static final String DELETE_ROOM_HISTORY =
         "DELETE FROM ofMucConversationLog WHERE roomID=?";
+    // Clear the chat history for a room but don't clear the messages that set the room's subject
```

Comment:
This is a good find. I'm wondering if there are other usages of `DELETE_ROOM_HISTORY` that instead should preserve the room subject.

Reply:
I haven't come across a use case yet for keeping a chatroom's topic after its deletion.